onnxruntime/test/contrib_ops/fused_conv_test.cc

```diff
@@ -29,93 +25,85 @@ struct ConvOpAndTestAttributes {
+  bool enable_cpu = (nullptr != DefaultCpuExecutionProvider().get()) && !use_float16 && !disable_cpu;
+
+  if (enable_cuda || enable_rocm || enable_cpu) {
+    OpTester test("FusedConv", 1, onnxruntime::kMSDomain);
+    test.AddAttribute("group", attributes.group);
+    test.AddAttribute("kernel_shape", attributes.kernel_shape);
+
+    if (!attributes.dilations.empty()) {
+      test.AddAttribute("dilations", attributes.dilations);
+    }
+
+    // Only one of pads / auto_pad can be present
+    if (!attributes.pads.empty()) {
+      test.AddAttribute("pads", attributes.pads);
+    } else {
+      test.AddAttribute("auto_pad", attributes.auto_pad);
+    }
+
+    if (!attributes.strides.empty()) {
+      test.AddAttribute("strides", attributes.strides);
+    }
+
+    ORT_ENFORCE(!attributes.activation.empty(), "activation must be set");
+    test.AddAttribute("activation", attributes.activation);
+
+    if (!attributes.activation_parameters.empty()) {
+      test.AddAttribute("activation_params", attributes.activation_parameters);
+    }
+
+    const char* szNames[] = {"X", "W", "B", "Z"};
+
+    if (use_float16) {
+      test.AddInput<MLFloat16>(szNames[0], input_shapes[0], ToFloat16(inputs[0]));
+      test.AddInput<MLFloat16>(szNames[1], input_shapes[1], ToFloat16(inputs[1]), weight_is_initializer);
+      if (inputs.size() >= 3)
+        test.AddInput<MLFloat16>(szNames[2], input_shapes[2], ToFloat16(inputs[2]));
+      if (inputs.size() >= 4)
+        test.AddInput<MLFloat16>(szNames[3], input_shapes[3], ToFloat16(inputs[3]));
+      test.AddOutput<MLFloat16>("Y", expected_output_shape, ToFloat16(expected_output));
+    } else {
+      test.AddInput<float>(szNames[0], input_shapes[0], inputs[0]);
+      test.AddInput<float>(szNames[1], input_shapes[1], inputs[1], weight_is_initializer);
+      if (inputs.size() >= 3)
+        test.AddInput<float>(szNames[2], input_shapes[2], inputs[2]);
+      if (inputs.size() >= 4)
+        test.AddInput<float>(szNames[3], input_shapes[3], inputs[3]);
+      test.AddOutput<float>("Y", expected_output_shape, expected_output);
+    }
+
+    if (enable_cuda) {
+      std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+      execution_providers.push_back(DefaultCudaExecutionProvider());
+      test.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, &execution_providers);
+    }
+
+    if (enable_rocm) {
+      std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+      execution_providers.push_back(DefaultRocmExecutionProvider());
+      test.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, &execution_providers);
+    }
+
+    if (enable_cpu) {
+      std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+      execution_providers.push_back(DefaultCpuExecutionProvider());
+      test.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, &execution_providers);
+    }
```

Comment:
Looks the test runs and verifies for every EP in `execution_providers` (https://github.com/microsoft/onnxruntime/blob/7ed3bfde513a7cad7ba8cc15e10ff82c0b030c94/onnxruntime/test/providers/provider_test_utils.cc#L1297-L1305). So this could be simplified as
```c++
    std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
    if (enable_cuda) {
      execution_providers.push_back(DefaultCudaExecutionProvider());
    }
    if (enable_rocm) {
      execution_providers.push_back(DefaultRocmExecutionProvider());
    }
    if (enable_cpu) {
      execution_providers.push_back(DefaultCpuExecutionProvider());
    }

    test.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, &execution_providers);
```